.github/workflows/main.yml

```diff
@@ -1,42 +1,188 @@
-name: Deploy GH Pages
-
-on:
-  push:
-    branches: 
-      - master 
-    paths:
-      - 'docs/*.md'
-      - 'docs/quickstart'
-      - '*.md'
-
-jobs:
-  deploy-gh-pages:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout master
-        uses: actions/checkout@v2
-        with:
-          path: master
-
-      - name: Checkout gh-pages branch
-        uses: actions/checkout@v2
-        with:
-          ref: 'gh-pages'
-          path: gh-pages
-    
-      - name: Copy site files
-        run: |
-          cp master/docs/*.md gh-pages/
-          cp -r master/docs/quickstart gh-pages/_pages
-          cp master/CODE_OF_CONDUCT.md master/CONTRIBUTING.md gh-pages/_pages/
-
-      - name: Commit GH Pages
-        run: |
-          cd gh-pages
-          git config user.name armada-admin
-          git config user.email admin@armadaproject.io
-          git add . _pages
-          git diff --quiet && git diff --staged --quiet || git commit -m "Updating Github Pages branch with latest Master changes"
-          git push
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+# Pre-merge TODO list
+# 1. change DOCKERTOKEN and DOCKERUSERNAME
+name: CI
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  GO111MODULE: "on"
+  GOPATH: "/home/runner/work/armada/armada/go"
+  GOCACHE: "/home/runner/.cache/go-build"
+
+on: 
+  push:
+    branches:
+      - '**'
+    pull_request:
+      branches:
+        - master
+    tags-ignore:
+      - '**'
+    paths-ignore:
+      - 'docs/*.md'
+      - 'docs/quickstart'
+      - '*.md'
+      - '**/release.yml'
+      - '**/pages.yml'
+
+jobs:
+  code_styles:
+    name: Code style
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+          
+      - name: Run ineffassign
+        run: |
+          cd /tmp
+          go install github.com/gordonklaus/ineffassign@latest
+          cd ${{ env.GOPATH }}/src/github.com/G-Research/armada
+          echo "Run ineffassign..."
+          ineffassign ./...
+
+      - name: Check formatting
+        run: |
+          go install golang.org/x/tools/cmd/goimports@v0.1.1

```

Comment:
imho `goimports` is more of a IDE tool which will automatically add imports, and format the code, if we need formatting only `gofmt` does the job and it doesen't need additional binaries

for full code quality I would go with `golint`, `gofmt`, and `go vet`, it will ensure that code style is good, formatting is proper, and do a code check for any potential issues

`golint` can be skipped if we don't want to be too strict tho, also we might have some lint issues already, worth checking

Reply:
thanks for the detailed explanainit! really appreciated 🙏 
when I pushed this version of workflow, we wanted to go with the same implementation as it was done with CircleCI [(current behavior)](https://github.com/G-Research/armada/blob/master/.circleci/config.yml#L136)
As I learned something new today, I'll see how `gofmt` behave with current implementation (e.g status code of `gofmt`) and try to push that as final decision